hooks/src/index.js

```diff
@@ -199,14 +199,30 @@ function flushAfterPaintEffects() {
 	afterPaintEffects = [];
 }
 
-function scheduleFlushAfterPaint() {
+options.scheduleFlushAfterPaint = () => {
 	setTimeout(flushAfterPaintEffects, 0);
+};
+
+export function act(cb) {
+	Component.__test__previousDebounce = options.debounceRendering;
+	options.debounceRendering = cb => Component.__test__drainQueue = cb;
+	const oldSchedule = options.scheduleFlushAfterPaint;
+	options.scheduleFlushAfterPaint = () => flushAfterPaintEffects();
+	const oldAfterPaint = options.afterPaint;
+	options.afterPaint = (component) => {
+		if (!component._afterPaintQueued && (component._afterPaintQueued = true) && afterPaintEffects.push(component) === 1) {
+			options.scheduleFlushAfterPaint();
+		}
+	};
+	cb();
+	options.afterPaint = oldAfterPaint;
+	options.scheduleFlushAfterPaint = oldSchedule;
 }
 
 if (typeof window !== 'undefined') {
-	afterPaint = (component) => {
+	options.afterPaint = (component) => {
 		if (!component._afterPaintQueued && (component._afterPaintQueued = true) && afterPaintEffects.push(component) === 1) {
-			requestAnimationFrame(scheduleFlushAfterPaint);
+			requestAnimationFrame(options.scheduleFlushAfterPaint);
```

Comment:
`requestAnimationFrame` is async too, similar to `setTimeout` it has slightly different scheduling semantings, but it's something we'd need to make synchronous too nonetheless. Maybe instead of swapping out the callback to `requestAnimationFrame` we can swap out it itself:

```js
(options.afterPaint || requestanimationFrame)(scheduleFlushAfterPaint);
```

That way we don't need to touch `scheduleFlushAfterPaint` at all.

Reply:
Correct me if I am wrong but scheduleAfterPaint introduces a setTimeout so we would have to override this aswell, no?

Reply:
It appears so, but by letting the user call the flushing function itself we can sidestep `scheduleFlushAfterPaint` completely :+1: